airflow/models/dag.py

```diff
@@ -912,12 +912,7 @@ def set_dag_runs_state(
             query = query.filter(DagRun.execution_date >= start_date)
         if end_date:
             query = query.filter(DagRun.execution_date <= end_date)
-        drs = query.all()
-
-        dirty_ids = []
-        for dr in drs:
-            dr.state = state
-            dirty_ids.append(dr.dag_id)
+        query.update({DagRun.state: state})
```

Comment:
Will this work if multiple DagRuns are returned?

Reply:
Yes, we are using this in many different places

Reply:
> > Can you just double-check & confirm that tests for this function are already covered?
> 
> @kaxil I've added a test for this method

Awesome, feel free to merge once the test passes